x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/find_rule_exceptions_route.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { transformError } from '@kbn/securitysolution-es-utils';
+import { getSavedObjectType } from '@kbn/securitysolution-list-utils';
+import { validate } from '@kbn/securitysolution-io-ts-utils';
+import type { FindResult } from '@kbn/alerting-plugin/server';
+
+import type { SecuritySolutionPluginRouter } from '../../../../types';
+import { DETECTION_ENGINE_RULES_URL } from '../../../../../common/constants';
+import { buildSiemResponse } from '../utils';
+import { enrichFilterWithRuleTypeMapping } from '../../rules/enrich_filter_with_rule_type_mappings';
+import type { FindExceptionReferencesOnRuleSchemaDecoded } from '../../../../../common/detection_engine/schemas/request/find_exception_list_references_schema';
+import { findExceptionReferencesOnRuleSchema } from '../../../../../common/detection_engine/schemas/request/find_exception_list_references_schema';
+import { buildRouteValidation } from '../../../../utils/build_validation/route_validation';
+import type { RuleReferencesSchema } from '../../../../../common/detection_engine/schemas/response/find_exception_list_references_schema';
+import { rulesReferencedByExceptionListsSchema } from '../../../../../common/detection_engine/schemas/response/find_exception_list_references_schema';
+import type { RuleParams } from '../../schemas/rule_schemas';
+
+export const findRuleExceptionReferencesRoute = (router: SecuritySolutionPluginRouter) => {
+  router.get(
+    {
+      path: `${DETECTION_ENGINE_RULES_URL}/exceptions/_find_references`,
+      validate: {
+        query: buildRouteValidation<
+          typeof findExceptionReferencesOnRuleSchema,
+          FindExceptionReferencesOnRuleSchemaDecoded
+        >(findExceptionReferencesOnRuleSchema),
+      },
+      options: {
+        tags: ['access:securitySolution'],
+      },
+    },
+    async (context, request, response) => {
+      const siemResponse = buildSiemResponse(response);
+
+      try {
+        const { ids, namespace_types: namespaceTypes, list_ids: listIds } = request.query;
+
+        const ctx = await context.resolve(['core', 'securitySolution', 'alerting']);
+        const rulesClient = ctx.alerting.getRulesClient();
+
+        if (ids.length !== namespaceTypes.length || ids.length !== listIds.length) {
```

Comment:
I agree that would be better, but I was trying to keep the call as a simple `GET` otherwise I'd need to change it to a `POST` right? In order to send through an array of objects. I don't anticipate this being an endpoint often used by users, mostly just devs.

Reply:
> I was trying to keep the call as a simple `GET` otherwise I'd need to change it to a `POST` right?

I haven't seen any limitations on the body of a `GET` request. I believe it could be anything.

> I don't anticipate this being an endpoint often used by users, mostly just devs.

Would it make sense to release that endpoint as `internal` then? That would allow us to introduce breaking changes to it freely if we decide to adjust the request/response schema.

You can find some examples of internal endpoints here:

https://github.com/elastic/kibana/blob/6a5c4b23c56cb886c7057d9c5117008022bb43b3/x-pack/plugins/security_solution/common/constants.ts#L289-L296

Reply:
Ah, good point. I'm going to update it to be internal.

I also do want to change it to the suggested array of objects - if it's ok, I'd like to do this as follow up as there's still lots more work to come.